Locale support for sa-currency-input and date-picker

Dastaan2k opened the pull request:
Added locale support for currency input and date time picker

Dastaan2k committed:
Added basic implementations of all components in the 'example' directory.

Dastaan2k committed:
added locale support to currency input and date time picker

Jeet-s commented:
remove package.json

Jeet-s commented on projects/simplified-ui/src/lib/sa-currency-input/sa-currency-input.component.ts:
```diff
@@ -1,75 +1,82 @@
 
+
+import localeIn from '@angular/common/locales/en-IN';
+registerLocaleData(localeIn);             /// Need to register the locale data before converting it to currency code.
```
register Spanish locale

Jeet-s commented on projects/simplified-ui/src/lib/sa-currency-input/sa-currency-input.component.ts:
```diff
@@ -1,75 +1,82 @@
   stateChanges = new Subject<void>();
   private _value: any;
-  private viewValue: string;
+  //private viewValue: string;
```
remove comment

Jeet-s commented on projects/simplified-ui/src/lib/sa-currency-input/sa-currency-input.component.ts:
```diff
@@ -1,75 +1,82 @@
+
+  constructor(private _focusMonitor: FocusMonitor, private _elementRef: ElementRef<HTMLElement>, @Optional() @Self() public ngControl: NgControl, @Inject(LOCALE_ID) private locale: string,private currencyPipe: CurrencyPipe) {
+
     this.decimalSeparator = '.';
```
get decimal separator based on locale

Jeet-s commented on src/app/app-routing.module.ts:
```diff
@@ -1,7 +1,33 @@
 import { NgModule } from '@angular/core';
```
make a different PR for examples

Jeet-s commented on projects/simplified-ui/src/lib/simplified-ui.module.ts:
```diff
@@ -146,6 +146,6 @@ import { SaPageHeaderComponent } from './sa-page-header/sa-page-header.component
   schemas: [CUSTOM_ELEMENTS_SCHEMA],
   entryComponents: [MatProgressSpinner],
-  providers: [{ provide: OWL_MOMENT_DATE_TIME_ADAPTER_OPTIONS, useValue: { useUtc: true } }]
+  providers: [{ provide: OWL_MOMENT_DATE_TIME_ADAPTER_OPTIONS, useValue: { useUtc: true } }, { provide: LOCALE_ID, useValue: 'en-IN' }]   // Override LOCALE_ID to test currency input for INR
```
we can revert this code and use the default locale

Dastaan2k committed:
- Added group selector for currency input.
- Both decimal and group selector values are now set based on the app's locale instead of hardcoded values.
- Registered spanish locale data.
- Removed the hardcoded locale from module.ts
- Removed commented code.

Dastaan2k committed:
- Removed log statements

Dastaan2k committed:
- Changed the apporach to get locale based decimal and group separator.

Dastaan2k committed:
- ..

Dastaan2k committed:
- Added examples

Dastaan2k committed:
Deleted package-lock.json

Dastaan2k committed:
- Added examples

Jeet-s commented on projects/simplified-ui/src/lib/sa-currency-input/sa-currency-input.component.ts:
```diff
@@ -1,73 +1,79 @@
   static nextId = 0;
-  private decimalSeparator: string;
+
+  private decimalSeparator;
```
add type

Jeet-s commented on projects/simplified-ui/src/lib/sa-currency-input/sa-currency-input.component.ts:
```diff
@@ -1,73 +1,79 @@
-  private decimalSeparator: string;
+
+  private decimalSeparator;
+  private groupSeparator;
```
same here

Dastaan2k committed:
- Removed extra files

Dastaan2k committed:
Merge branch 'locale-support-new' of https://github.com/Dastaan2k/simplified-ui into locale-support-new

Dastaan2k committed:
- Removed extra files ...

Dastaan2k committed:
- Reverted allowNegativeValue.
- Fixed shouldLabelFloat based on current focused state and FormControl's value

Jeet-s commented on projects/simplified-ui/src/lib/sa-currency-input/sa-currency-input.component.ts:
```diff
@@ -1,73 +1,81 @@
+
+import localeEs from '@angular/common/locales/es';
+
+registerLocaleData(localeEs);
```
we can register locale in the app only

Jeet-s commented on projects/simplified-ui/src/lib/sa-currency-input/sa-currency-input.component.ts:
```diff
@@ -80,10 +88,7 @@ export class SaCurrencyInputComponent
-      this.viewValue = num;
-      this.value = this.parse(num);
-    });
+    this.currencyValue.valueChanges.pipe(debounceTime(200)).subscribe((num) => {});
```
we can remove this subscription

Jeet-s commented on projects/simplified-ui/src/lib/sa-currency-input/sa-currency-input.component.ts:
```diff
@@ -142,7 +151,7 @@ export class SaCurrencyInputComponent
   set value(val: string | null) {
     this._value = val;
-    if (this._value != null && this._value !== '') {
+    if (this._value != null && this._value !== '' && this._value >= 0) {
```
remove greater than check

Jeet-s commented on projects/simplified-ui/src/lib/sa-currency-input/sa-currency-input.component.ts:
```diff
@@ -103,7 +108,10 @@ export class SaCurrencyInputComponent
   get shouldLabelFloat() {
-    return this.focused || !this.empty;
+    if (!this.focused) {
+      return this.currencyValue.value !== null;
```
can we use this.empty instead of checking null?

Jeet-s commented on projects/simplified-ui/src/lib/sa-currency-input/sa-currency-input.component.ts:
```diff
@@ -1,73 +1,81 @@
   @ViewChild('input') inputRef: ElementRef;
-  @Input() allowNegative: boolean = true;
+
+  @Input()allowNegative: boolean = true;
```
add space

Jeet-s commented on projects/simplified-ui/src/lib/sa-currency-input/sa-currency-input.component.ts:
```diff
@@ -1,73 +1,81 @@
+
+      if(this.currencyValue.value != null) {
+        if (origin) {
+          this.currencyValue.setValue(this.parse(this.currencyValue.value.replace(getLocaleCurrencySymbol(this.locale), '')));
```
we should add logic to replace currency symbol in parse method

Jeet-s commented on angular.json:
```diff
@@ -156,12 +173,20 @@
-}
+  "defaultProject": "simplified-ui-library",
+  "cli": {
+    "analytics": "cb8df5c8-539c-4c04-bbee-3864e14c6b3a"
```
why are we using this analytics property?

Dastaan2k committed:
- Completed remaining changes

Dastaan2k committed:
- Completed remaining changes

Dastaan2k committed:
- Removed example related components and package-lock and angular.json

Dastaan2k committed:
- First numerical value initialization bug through form control for currency input
